src/components/autocomplete/js/autocompleteController.js

```diff
@@ -441,12 +441,21 @@ function MdAutocompleteCtrl ($scope, $element, $mdUtil, $mdConstant, $mdTheming,
         select(ctrl.index);
         break;
       case $mdConstant.KEY_CODE.ESCAPE:
+        // Only return if blur is disabled
+        // Only return if dropdown is hidden and not loading
+        // Only return if text clearing is disabled or text is already cleared
+        if (!!$scope.noBlurOnEscape && ctrl.hidden && !ctrl.loading && (!!$scope.noClearOnEscape || !$scope.searchText))
+          return; 
+
         event.stopPropagation();
         event.preventDefault();
-        if ($scope.searchText) clearValue();
 
-        // Force the component to blur if they hit escape
-        doBlur(true);
+        clearValue(!$scope.noClearOnEscape && $scope.searchText);
+
+        if (!$scope.noBlurOnEscape)
```

Comment:
Please use braces here and with the `else` to follow our [Style Guide](https://google.github.io/styleguide/javascriptguide.xml?showone=Code_formatting#Code_formatting).